### PR TITLE
Refine layernorm naming and test code

### PR DESCRIPTION
**Pull request by rocking5566**
1. Use normalization instead of layernorm inside of the kernel
2. Use profiler in test code
3. Add parameter to saveMean and InvVar for future backward pass

### CODE REVIEW

File: profiler/include/profile_layernorm_impl.hpp [hunk 1]
```diff
@@ -184,20 +194,32 @@ void profile_layernorm_impl(int do_verification,
...
+        LogRange(std::cout << "length = ", length, ",") << ", ";
+        LogRange(std::cout << "stride = ", strideXY, ",") << ", ";
+        LogRange(std::cout << "reduce dims ", reduce_dim, ",") << std::endl;
+        std::cout << "best perf = " << best_avg_time << " ms, " << best_gb_per_sec << " GB/s, "
+                  << best_instance_name << std::endl;
+    }
+
+    if(num_kernel == 0)
+    {
+        std::cout << "Error: No kernel is tested" << std::endl;
```
Comment (shaojiewang) [inline profiler/include/profile_layernorm_impl.hpp:218]: Does it mean that the case is not applicable to this op? I think ```not applicable``` hint may be better log.

File: test/normalization/test_layernorm2d_fp16.cpp [hunk 2]
```diff
@@ -2,28 +2,44 @@
...
+    using GammaDataType = std::tuple_element_t<1, Tuple>;
+    using BetaDataType  = std::tuple_element_t<2, Tuple>;
+    using AccDataType   = std::tuple_element_t<3, Tuple>;
+    using YDataType     = std::tuple_element_t<4, Tuple>;
+
+    void Run()
+    {
+        // N, H, W, G, C
+        std::vector<std::vector<ck::index_t>> lengths = {
+            {4, 256}, {8, 511}, {9, 1032}, {4, 2048}, {1, 8192}, {4000, 2000}};
```
Comment (shaojiewang) [inline test/normalization/test_layernorm2d_fp16.cpp:25]: Question: for a single case, ```{4, 256}```, which number is ```N H W G C```?

**Review by shaojiewang — COMMENTED** [review]
Code almost looks good to me. Thanks for re-naming and adding tests. Just 2 questions.

File: profiler/include/profile_layernorm_impl.hpp (hunk 1 above)
Comment (rocking5566) [reply profiler/include/profile_layernorm_impl.hpp]: Agree. I fix the string both layernorm and groupnorm in 33d760fc8adae96b3134a63516f282beba279aab

File: test/normalization/test_layernorm2d_fp16.cpp (hunk 2 above)
Comment (rocking5566) [reply test/normalization/test_layernorm2d_fp16.cpp:25]: It is just 2D. I fix the comment in 33d760fc8adae96b3134a63516f282beba279aab

**Review by shaojiewang — APPROVED** [review]
LGTM

**Review by shaojiewang — APPROVED** [review]

**Pull request merged**